### ISSUE TITLE
TensorBoard not syncing with wandb

### ISSUE THREAD

**Issue by sff1019** [wandb/client#1782]
**Describe the bug**
Tensorboard is not being synced with wandb, even when specifying `wandb.init(sync_tensorboard=True)`

**To Reproduce**
1. Initialize wandb: `wandb.init(sync_tensorboard=True)`
2. Log to tensorboard using `tf.summary.scalar()`

Simple TensorFlow training script to reproduce this manner:
https://gist.github.com/sff1019/8870a9e15e95b9e469cd3419b7891c2e

**Expected behavior**
It was expected to log metrics by syncing TensorBoard (Followed the directions as shown here: https://docs.wandb.ai/integrations/tensorboard)

**Screenshots**
As shown in the screenshot, no metric / TensorBoard was logged.
![Screen Shot 2021-02-01 at 15 28 17](https://user-images.githubusercontent.com/31456798/106422538-1e412f00-64a2-11eb-9068-45bbf09c2904.png)

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04.1
 - Browser: Chrome
 - Version:
   - wandb==0.10.15
   - tensorflow==2.3.0
   - tensorboard==2.4.1
   
**Additional context**
I haven't tested with logging into TensorBoard using `tensorboard_callback = tf.keras.callbacks.TensorBoard(log_dir=log_dir, histogram_freq=1)`.

**Comment by ariG23498** [comment]
Hey @sff1019 
Thanks for a great repro along with the ticket
I would add to the issue here that
```python
# Sync TensorBoard with WandB.
wandb.tensorboard.patch(root_logdir=train_log_dir)
wandb.init(entity="xxx", project="xxx")
```
Aslo fails to track the tfevent files


Tagging @vanpelt  for some expertise in this ticket.

**Comment by vanpelt** [comment]
@sff1019 can you share how you're logging to tensorboard in your script if you aren't using the keras tensorboard callback?

**Comment by vanpelt** [comment]
@sff1019 never mind, I see your gist now.  We'll take a look ASAP.

**Comment by TusharNimbhorkar** [comment]
@vanpelt @ariG23498 @sff1019  I am having same issue with TensorboardX #1696

**Comment by aj7tesh** [comment]
same issue. Anyone could find a solution to it

**Comment by ariG23498** [comment]
Hey folks,
Just a heads up, and we’re going to emphasize this in the docs, but you have to create the `file_writers` after calling `wandb.init`

Let us know if this helped! 😄

**Comment by sff1019** [comment]
@ariG23498 Thank you for the info! This seems to work! 😄

**Comment by aj7tesh** [comment]
@sff1019 could you explain

**Comment by ariG23498** [comment]
> Simple TensorFlow training script to reproduce this manner:
> https://gist.github.com/sff1019/8870a9e15e95b9e469cd3419b7891c2e
In this training script I have modified the following:
```python
# Sync TensorBoard with WandB.
wandb.tensorboard.patch(root_logdir='logs/gradient_tape/')
wandb.init(entity="xxx", project="xxx")

current_time = datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
train_log_dir = 'logs/gradient_tape/' + current_time + '/train'
test_log_dir = 'logs/gradient_tape/' + current_time + '/test'
train_summary_writer = tf.summary.create_file_writer(train_log_dir)
test_summary_writer = tf.summary.create_file_writer(test_log_dir)

model = create_model()

EPOCHS = 5

for epoch in range(EPOCHS):
    for (x_train, y_train) in train_dataset:
        train_step(model, optimizer, x_train, y_train)
    with train_summary_writer.as_default():
        tf.summary.scalar('loss', train_loss.result(), step=epoch)
        tf.summary.scalar('accuracy', train_accuracy.result(), step=epoch)

    for (x_test, y_test) in test_dataset:
        test_step(model, x_test, y_test)
    with test_summary_writer.as_default():
        tf.summary.scalar('loss', test_loss.result(), step=epoch)
        tf.summary.scalar('accuracy', test_accuracy.result(), step=epoch)

    template = 'Epoch {}, Loss: {}, Accuracy: {}, Test Loss: {}, Test Accuracy: {}'
    print(
        template.format(epoch + 1, train_loss.result(),
                        train_accuracy.result() * 100, test_loss.result(),
                        test_accuracy.result() * 100))

    # Reset metrics every epoch
    train_loss.reset_states()
    test_loss.reset_states()
    train_accuracy.reset_states()
    test_accuracy.reset_states()
```

I am closing this issue for the time being. Please let us know if the problem persists! :smile:

**Comment by ringohoffman** [comment]
Very relevant: 

* https://github.com/wandb/wandb/issues/1782#issuecomment-779161203

> Just a heads up, and we’re going to emphasize this in the docs, but you have to create the file_writers after calling `wandb.init`
